lib/Common/Common/NumberUtilities.h

```diff
@@ -115,6 +115,62 @@ namespace Js
             FormatPrecision
         };
 
+        // Calculates magic number (multiplier) and shift amounts (shiftAmt) to replace division by constants with multiplication and shifts
+        // Ref: Warren's Hacker's Delight, Chapter 10.
+        struct DivMagicNumber
+        {
+            DivMagicNumber(uint64 multiplier, uint shiftAmt) : multiplier(multiplier), shiftAmt(shiftAmt) {}
+            int64 multiplier;
```

Comment:
looks like this only needs to be an int32?

Reply:
yes. done.